packages/backend/server/src/core/doc-service/job.ts

```diff
@@ -1,61 +1,34 @@
-import { Injectable, Logger, OnModuleInit } from '@nestjs/common';
-import { SchedulerRegistry } from '@nestjs/schedule';
-import { CLS_ID, ClsService } from 'nestjs-cls';
+import { Injectable, Logger } from '@nestjs/common';
 
-import { CallMetric, Config, genRequestId, metrics } from '../../base';
+import { metrics, OnJob } from '../../base';
 import { PgWorkspaceDocStorageAdapter } from '../doc';
 
+declare global {
+  interface Jobs {
+    'doc.mergePendingDocUpdates': {
+      workspaceId: string;
+      docId: string;
+    };
+  }
+}
+
 @Injectable()
-export class DocServiceCronJob implements OnModuleInit {
-  private busy = false;
+export class DocServiceCronJob {
   private readonly logger = new Logger(DocServiceCronJob.name);
 
-  constructor(
-    private readonly config: Config,
-    private readonly cls: ClsService,
-    private readonly workspace: PgWorkspaceDocStorageAdapter,
-    private readonly registry: SchedulerRegistry
-  ) {}
-
-  onModuleInit() {
-    if (this.config.doc.manager.enableUpdateAutoMerging) {
-      this.registry.addInterval(
-        this.autoMergePendingDocUpdates.name,
-        // scheduler registry will clean up the interval when the app is stopped
-        setInterval(() => {
-          if (this.busy) {
-            return;
-          }
-          this.busy = true;
-          this.autoMergePendingDocUpdates()
-            .catch(() => {
-              /* never fail */
-            })
-            .finally(() => {
-              this.busy = false;
-            });
-        }, this.config.doc.manager.updatePollInterval)
-      );
+  constructor(private readonly workspace: PgWorkspaceDocStorageAdapter) {}
 
-      this.logger.log('Updates pending queue auto merging cron started');
+  @OnJob('doc.mergePendingDocUpdates')
+  async mergePendingDocUpdates({
+    workspaceId,
+    docId,
+  }: Jobs['doc.mergePendingDocUpdates']) {
+    try {
+      await this.workspace.getDoc(workspaceId, docId);
+    } catch (e) {
+      metrics.doc.counter('auto_merge_pending_doc_updates_error').add(1);
+      this.logger.error('Failed to auto merge pending doc updates', e);
+      throw e;
     }
   }
-
-  @CallMetric('doc', 'auto_merge_pending_doc_updates')
-  async autoMergePendingDocUpdates() {
-    await this.cls.run(async () => {
-      this.cls.set(CLS_ID, genRequestId('job'));
-      try {
-        const randomDoc = await this.workspace.randomDoc();
-        if (!randomDoc) {
-          return;
-        }
-
-        await this.workspace.getDoc(randomDoc.workspaceId, randomDoc.docId);
```

Comment:
No need to schedule tasks for merging updates? Rely entirely on user requests to trigger them?

Reply:
there won't be any updates if user not sending any